common/utils/filters.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  CatalogueAggregationBucket,
+  CatalogueAggregations,
+} from '@weco/common/model/catalogue';
+import { nodeListValueToArray, inputValue } from '@weco/common/utils/forms';
+import { LinkProps } from 'next/link';
+
+import {
+  worksLink,
+  imagesLink,
+} from '@weco/common/services/catalogue/ts_routes';
+
+export const getFilterByCount = (
+  bucketFilter: CatalogueAggregationBucket[]
+): CatalogueAggregationBucket[] => {
+  return bucketFilter.filter(items => {
+    return items.count > 0;
+  });
+};
+
+export const getAggregationFilterByName = (
+  aggregations: CatalogueAggregations | undefined,
+  name: string
+): CatalogueAggregationBucket[] | [] => {
+  return aggregations && aggregations?.[name]?.buckets
+    ? getFilterByCount(aggregations?.[name]?.buckets)
+    : [];
+};
+
+type RadioGroupOption = {
+  value: string;
+  id: string;
+  label: string;
+};
+
+// sort by the highest count
+export const sortByCount = (
+  a: CatalogueAggregationBucket,
+  b: CatalogueAggregationBucket
+): number => {
+  if (a.count > b.count) {
+    return -1;
+  } else {
+    return 1;
+  }
+};
+
+export const getAggregationRadioGroup = (
+  aggregation: CatalogueAggregationBucket[],
+  prefixId: string
+): RadioGroupOption[] => {
+  const radioOption = aggregation.sort(sortByCount).map(item => {
+    const id = item?.data?.label.replace(/\s/, '-').toLocaleLowerCase();
+    return {
+      value: item?.data?.label,
+      id: `${prefixId}-${id}`,
+      label: `${item?.data?.label} (${item?.count})`,
+    };
+  });
+
+  return radioOption;
+};
+
+export const getFilterItemSelected = (
+  form: HTMLFormElement,
+  formName: string
```

Comment:
I'm assuming this isn't the `formName` but rather the name of group of elements you'd like to search through?

Does this return a `string[]` of values of the selected elements in a checkbox radio group?

Reply:
ahh yes the naming attribute is wrong , yeah its the input element within the form. 

The checkbox radio i need to double check, but the checkbox items returns string[]